<launch>

    <node name="image_republish_fpv" pkg="image_transport" type="republish" args="raw compressed">
        <remap from="in" to="/dji_osdk_ros/fpv_camera_images" />
        <remap from="out" to="/dji_osdk_ros/fpv_camera_images/compressed" />
    </node>

    <node name="image_republish_main" pkg="image_transport" type="republish" args="raw compressed">
        <remap from="in" to="/dji_osdk_ros/main_camera_images" />
        <remap from="out" to="/dji_osdk_ros/main_camera_images/compressed" />
    </node>
    
    <node pkg="gk_drone_pkg" type="yolo_to_tf" name="yolo_to_tf" output="screen">
        <remap from="msg_yolo" to="yolov7/yolov7"/>
        <remap from="image_row_yolo" to="yolov7/yolov7/visualization"/>
        <remap from="dji_osdk_ros/gps_position" to="dji_osdk_ros/gps_position"/>
        <remap from="dji_osdk_ros/gimbal_angle" to="dji_osdk_ros/gimbal_angle"/>
        <remap from="dji_osdk_ros/imu" to="dji_osdk_ros/imu"/>
        <param name="drone_tf" type="string" value="drone"/>
        <param name="camera_tf" type="string" value="camera"/>
        <param name="map_tf" type="string" value="nav"/>
        <param name="file_path" type="string" value="/home/dji/osdk_ros_ws/src/gk_drone_pkg/script/test_10000.txt"/>
        <param name="origin_Lat"  value="43.28758036"/>
        <param name="origin_Lon"  value="143.1873489"/>
        <param name="thresh_humandetection" value="10.0"/>
        <param name="magin_high"  value="100"/>
        <param name="magnification"  value="1.0"/>
        <param name="write_file_path" type="string" value="/home/dji/Dropbox/JIC2022"/>
    </node>

    <!--node pkg="gk_drone_pkg" type="google_drive_test.py" name="google_drive_test" output="screen">
    </node-->  

    <node pkg="rosbag" type="record" name="drone_bag_recorder" output="screen"
          args= "record -o /home/dji/osdk_ros_ws/src/gk_drone_pkg/rosbag/
                 /dji_osdk_ros/attitude /dji_osdk_ros/flight_status /dji_osdk_ros/main_camera_images/compressed/compressed 
                 /dji_osdk_ros/fpv_camera_images/compressed/compressed
                 /dji_osdk_ros/gimbal_angle  /dji_osdk_ros/gps_health /dji_osdk_ros/gps_position
                 /dji_osdk_ros/height_above_takeoff /dji_osdk_ros/imu /dji_osdk_ros/local_frame_ref
                 /dji_osdk_ros/local_position /dji_osdk_ros/vo_position /dji_osdk_ros/velocity"/> 


</launch>
